### PR TITLE
Feat: consolidate our test matrix pattern

### PR DESCRIPTION
**Pull request by TomerHeber**
One folder for all tests.
Each version has a configuration file instead (no longer contains tests).

### CODE REVIEW

File: terratag_test.go [hunk 1]
```diff
@@ -23,14 +26,19 @@ var args = []string{
 	programName,
 	"-tags={\"env0_environment_id\":\"40907eff-cf7c-419a-8694-e1c6bf1d1168\",\"env0_project_id\":\"43fd4ff1-8d37-4d9d-ac97-295bd850bf94\"}",
 }
-var rootDir = "test/fixture"
+var testsDir = "test/tests"
```
Comment (TomerHeber) [inline terratag_test.go:29]: tests directory contains all availabel tests.

File: terratag_test.go [hunk 2]
```diff
@@ -23,14 +26,19 @@ var args = []string{
 	programName,
 	"-tags={\"env0_environment_id\":\"40907eff-cf7c-419a-8694-e1c6bf1d1168\",\"env0_project_id\":\"43fd4ff1-8d37-4d9d-ac97-295bd850bf94\"}",
 }
-var rootDir = "test/fixture"
```
Comment (TomerHeber) [inline terratag_test.go:26]: fixture directory contains  terreform versions, and each version has a config file.
I deleted all the tests from this directories and instead added a config file. (see config.yaml for an example).

File: terratag_test.go [hunk 3]
```diff
@@ -134,24 +142,55 @@ func itShouldTerraformInit(entryDir string, g *GomegaWithT) {
 	g.Expect(err).To(BeNil(), "terraform init failed")
 }
 
-func getEntries(version string) []TestCase {
+func getConfig(terraformDir string) (*TestCaseConfig, error) {
```
Comment (TomerHeber) [inline terratag_test.go:145]: This function is for getting the config for each version.

File: terratag_test.go [hunk 4]
```diff
@@ -134,24 +142,55 @@ func itShouldTerraformInit(entryDir string, g *GomegaWithT) {
...
 		entryDir := strings.TrimSuffix(slashed, "/main.tf")
 		terraformPathSplit := strings.Split(slashed, terraformDir)
-		pathAfterTerraformDir := terraformPathSplit[1]
-		suite := strings.Split(pathAfterTerraformDir, "/")[1]
 		pathBeforeTerraformDir := terraformPathSplit[0]
-		suiteDir := pathBeforeTerraformDir + terraformDir + "/" + suite
+		suiteDir := pathBeforeTerraformDir + terraformDir + "/main.tf"
+		suite := strings.Split(pathBeforeTerraformDir, "/")[2]
+
+		if _, ok := suitesMap[suite]; !ok {
```
Comment (TomerHeber) [inline terratag_test.go:190]: skip the test if not listed in config file.

File: terratag_test.go [hunk 5]
```diff
@@ -163,9 +202,9 @@ func getEntries(version string) []TestCase {
 	return testEntries
 }
 
-func cloneOutput(inputDirs []string) {
+func cloneOutput(inputDirs []string, terraformDir string) {
 	for _, inputDir := range inputDirs {
-		outputDir := strings.TrimSuffix(inputDir, "input") + "out"
+		outputDir := strings.TrimSuffix(inputDir, "input") + "out" + terraformDir
```
Comment (TomerHeber) [inline terratag_test.go:207]: just to prevent overriding on each other, each version clones to out + /terraformdir

File: terratag_test.go [hunk 6]
```diff
@@ -195,12 +234,21 @@ func terratag(entryDir string, filter string) (err interface{}) {
 
 func terraform(entryDir string, cmd string) error {
 	println("terraform", cmd)
+	var stdout bytes.Buffer
```
Comment (TomerHeber) [inline terratag_test.go:237]: errors were not shown in a readable manner. This fixes it.

File: test/fixture/terraform_11/config.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+suites:
```
Comment (TomerHeber) [inline test/fixture/terraform_11/config.yaml:1]: here's an example...

**Comment by TomerHeber** [comment]
Note: 224 files changed is not accurate. It's mostly deleted files. I also moved tests from one directory to the other.
In any case, the best way to review this is to ignore any ".tf" files (moved or deleted).

File: go.mod [hunk 8]
```diff
@@ -4,57 +4,74 @@ go 1.18
...
 require (
 	github.com/bmatcuk/doublestar v1.2.2
-	github.com/hashicorp/go-hclog v0.9.2
+	github.com/hashicorp/go-hclog v1.2.0
 	github.com/hashicorp/hcl/v2 v2.8.0
 	github.com/hashicorp/logutils v1.0.0
 	github.com/minamijoyo/tfschema v0.6.0
 	github.com/onsi/gomega v1.10.1
 	github.com/otiai10/copy v1.2.0
+	github.com/spf13/viper v1.11.0
```
Comment (TomerHeber) [inline go.mod:13]: Viper is a popular (19k stars) golang library for managing configuration files.

**Review by sagilaufer1992 — APPROVED** [review]
looks great! neat organizing

**Pull request merged**